### PR TITLE
feat(observability): capture HTTP error response body in outbound clients

### PR DESCRIPTION
**Pull request by pannpers**
## Summary

- **New**: `pkg/httpx.CaptureResponseBody` helper captures up to 1024 bytes of error response body, sanitizes non-printable chars, and drains remainder for connection reuse
- **Fix**: `pkg/api.FromHTTP` now attaches captured body as `slog.Attr("responseBody")` — covers Google Maps, fanart.tv, Last.fm, MusicBrainz clients (4 call sites)
- **Fix**: `webpush/sender.go` and `fanarttv/logo_fetcher.go` apply same capture in their custom error paths (preserving 410/404 semantics)
- **Docs**: Debug RPC runbook updated with "How to verify delivery succeeded" section + PGA troubleshooting table

Part of OpenSpec change `unblock-fcm-via-private-vip`. Ships independently of the cloud-provisioning PGA VIP switch.

## Motivation

During FCM 403 investigation, the decisive diagnostic body ("Your client does not have permission to get URL ... from this server") was invisible in production logs because all outbound HTTP clients discarded error bodies. Hours of manual investigation were required to surface it via external CLI replication.

## Test plan

- [x] `pkg/httpx/body_test.go`: 9 boundary cases (empty, truncation, binary, reader error)
- [x] `pkg/api/errors_test.go`: 4 cases (200 nil, 403+body, 500 empty)
- [x] `webpush/sender_test.go`: +2 cases (400+body, 410 no body attr)
- [x] `fanarttv/logo_fetcher_test.go`: +1 case (403+body)
- [x] `make lint` passes
- [ ] CI (Lint, Test, Vulnerability Check, claude-review)

Refs: #281

### CODE REVIEW

File: pkg/httpx/body.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+		case c == '\t' || c == '\n' || c == '\r':
+			out.WriteByte(c)
+		case c < 0x20 || c >= 0x7F:
+			out.WriteRune('\uFFFD')
+		default:
+			out.WriteByte(c)
+		}
+	}
+	return out.String()
+}
```
[the remark this answers is not in the document]
Comment (pannpers) [reply pkg/httpx/body.go:65]: Fixed in e04c9d9 — switched to range-over-string iteration which auto-decodes UTF-8 runes. Only control characters and invalid byte sequences are now replaced. Added 2 test cases: valid Japanese multi-byte string + mixed valid UTF-8 with control characters.
